arch/risc-v/src/mpfs/mpfs_entrypoints.c

```diff
@@ -53,71 +54,56 @@ static uint64_t g_app_entrypoints[] =
   CONFIG_MPFS_HART4_ENTRYPOINT
 };
 
-/****************************************************************************
- * Private Functions
- ****************************************************************************/
-
-static void jump_to_app(void) naked_function;
-static void jump_to_app(void)
-{
-  __asm__ __volatile__
-    (
-      "csrr a0, mhartid\n"           /* Hart ID */
-      "slli t1, a0, 3\n"             /* To entrypoint offset */
-      "la   t0, g_app_entrypoints\n" /* Entrypoint table base */
-      "add  t0, t0, t1\n"            /* Index in table */
-      "ld   t0, 0(t0)\n"             /* Load the address from table */
-      "jr   t0"                      /* Jump to entrypoint */
-    );
-}
-
-/****************************************************************************
- * Public Data
- ****************************************************************************/
-
-/* Default boot address for every hart */
-
-extern void mpfs_opensbi_prepare_hart(void);
-
-/* Trampoline functions, jump to SBI if so configured, to app if not */
-
-const uint64_t g_entrypoints[5] =
-{
-#ifdef CONFIG_MPFS_HART0_SBI
-  (uint64_t)mpfs_opensbi_prepare_hart,
-#else
-  (uint64_t)jump_to_app,
-#endif
-
+static uint64_t g_hart_use_sbi =
 #ifdef CONFIG_MPFS_HART1_SBI
-  (uint64_t)mpfs_opensbi_prepare_hart,
-#else
-  (uint64_t)jump_to_app,
+  (1 << 1) |
 #endif
 
 #ifdef CONFIG_MPFS_HART2_SBI
-  (uint64_t)mpfs_opensbi_prepare_hart,
-#else
-  (uint64_t)jump_to_app,
+  (1 << 2) |
 #endif
 
 #ifdef CONFIG_MPFS_HART3_SBI
-  (uint64_t)mpfs_opensbi_prepare_hart,
-#else
-  (uint64_t)jump_to_app,
+  (1 << 3) |
 #endif
 
 #ifdef CONFIG_MPFS_HART4_SBI
-  (uint64_t)mpfs_opensbi_prepare_hart,
-#else
-  (uint64_t)jump_to_app,
+  (1 << 4) |
 #endif
-};
+  0;
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+extern void mpfs_opensbi_prepare_hart(void);
 
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
 
+void __jump_to_app(void) naked_function;
+void __jump_to_app(void)
+{
+  __asm__ __volatile__
+    (
+      "csrr a0, mhartid\n"                   /* Hart ID */
+      "ld   t0, g_hart_use_sbi\n"
```

Comment:
As I commented the inline code it would be polite to comment these as well, even though they are obvious to someone who knows rv assembly.